src/report/highlight.rs

```diff
@@ -126,9 +133,112 @@ mod imp {
 mod imp {
     use super::*;
 
-    pub(crate) fn create_highlighter(_color: bool) -> Result<Box<dyn Highlighter>> {
-        // No-op highlighter for wasm
-        return Ok(Box::new(NopHighlighter))
+    use tree_sitter_bpf_c::LANGUAGE;
+    use tree_sitter_highlight::Highlight;
+    use tree_sitter_highlight::HighlightConfiguration;
+    use tree_sitter_highlight::HighlightEvent;
+    use tree_sitter_highlight::Highlighter as TsHighlighter;
+
+
+    struct TreeSitterHtmlHighlighter {
+        highlight_config: HighlightConfiguration,
+    }
```

Comment:
In my opinion this is too much of a duplication of the existing `TreeSitterHighlighter`. Can we share the  core of the implementation instead of duplicating it?